core/engine/src/builtins/promise/mod.rs

```diff
@@ -462,6 +463,57 @@ impl Promise {
         &self.state
     }
 
+    /// [`Promise.try ( callbackfn, ...args )`][spec]
+    ///
+    /// Calls the given function and returns a new promise that is resolved if the function
+    /// completes normally and rejected if it throws.
+    ///
+    /// [spec]: https://tc39.es/proposal-promise-try/#sec-promise.try
+    pub(crate) fn r#try(
+        this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        let callback = args.get_or_undefined(0);
+        let callback_args = args.get(1..).unwrap_or(&[]);
+
+        // 1. Let C be the this value.
+        // 2. If C is not an Object, throw a TypeError exception.
+        let c = this.as_object().ok_or_else(|| {
+            JsNativeError::typ().with_message("Promise.try() called on a non-object")
+        })?;
+
+        // 3. Let promiseCapability be ? NewPromiseCapability(C).
+        let promise_capability = PromiseCapability::new(c, context)?;
+
+        // 4. Let status be Completion(Call(callbackfn, undefined, args)).
+        let status = callback.call(&JsValue::undefined(), callback_args, context);
+
+        // 5. If status is an abrupt completion, then
+        if let Err(e) = status {
+            let value = e.to_opaque(context);
+
+            // a. Perform ? Call(promiseCapability.[[Reject]], undefined, « status.[[Value]] »).
+            promise_capability
+                .functions
+                .reject
+                .call(&JsValue::undefined(), &[value], context)?;
+        }
+        // 6. Else,
+        else {
+            let value = status.unwrap();
+
+            // a. Perform ? Call(promiseCapability.[[Resolve]], undefined, « status.[[Value]] »).
+            promise_capability
+                .functions
+                .resolve
+                .call(&JsValue::undefined(), &[value], context)?;
+        }
```

Comment:
We could avoid the unwrap with a match, since we use both `Ok` and `Err` values.

```rust
match status {
   // 5. If status is an abrupt completion, then
   Err(err) => { .. }
   // 6. Else,
   Ok(value) => { .. }
}
```

Reply:
Thanks, will fix this up later today! Pretty much my first time writing rust 😄